### ISSUE TITLE
template feature: handle inline block of text (replace section of the file)

### ISSUE THREAD

**Issue by ghost** [ansible/ansible-modules-core#2441]
template, lineinfile, replace, patch modules are great but not 
enough when you want to add multiple lines of text to a file 
without replacing the whole file (not under full ansible control) 
you can not do it easily with existing modules.

Just search for "lineinfile multiple lines" on your search engine...

The feature idea is to implement the following in template module
- inline: yes|no
- before: regexp (mandatory if inline is yes)
- after: regexp (mandatory if inline is yes)

If inline equal to yes you evaluate the output of the template
and replace the content of the file between before regexp and
after regexp marks (those reference lines excluded) with the
new output of the template.

So you can use several templates from several roles to complete
the content of the same file and also share the control of the file 
with other applications, automation, editor etc (not like template as-is)

To sum up I would like the fusion of lineinfile and template in one
powerful module!

The closer module to that is blockinfile but it does not include
the features of template (variables, filters, etc.)
https://github.com/yaegashi/ansible-role-blockinfile

Applications:
- Add multiple hosts computed with filters to /etc/hosts
- Add multiple interfaces computed with filters to /etc/network/interfaces
- Add multiple mount points from template to /etc/fstab without touching existing lines
- Add ntp servers list from template to /etc/openntpd/ntpd.conf or /etc/ntp.conf keeping the rest as-is
- Add more new security rules to /etc/apache2/conf-available/security.conf
- Add a new exception to apparmor ex. /etc/apparmor.d/abstractions/libvirt-qemu
- Add more "Require ip " from template to /etc/apache2/mods-available/status.conf
- Add text with TABULATIONs from the template into a file without ugly hack {{'\t'}} (aka /etc/hosts)
- Remove block of text between before and after regexp if content of template is empty
- Generate a single config extensions.conf file from multiple roles for asterisk (IVR, Extentions, etc.)
- Add to existing DNS zone a section with dynamic generated DNS records from template
- Add to existing Cisco configuration a section to handle ACLs
- ...

REFERENCES
https://github.com/ansible/ansible-modules-core/issues/557
https://github.com/ansible/ansible/issues/5679
http://stackoverflow.com/questions/24334115/ansible-lineinfile-for-several-lines
http://comments.gmane.org/gmane.comp.sysutils.ansible/6359
https://brokenbad.com/loops-with-lineinfile-ansible-module/
https://github.com/ansible/ansible/issues/4818
https://github.com/ansible/ansible/issues/4710

**Comment by robynbergeron** [comment]
@lordbaco --  Thanks for submitting this feature idea. Since the blockinfile module has now been merged into extras (thanks, @yaegashi!) -- could you open this issue in extras so it can be a feature/enhancement request for that module?

**Comment by ansibot** [comment]
@lordbaco closure has been requested for this issue or it has timed out waiting for your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)
